modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -859,19 +1020,60 @@ private static boolean invalid(RocksIterator it) {
      * Converts raw byte array representation of the value into a binary row.
      *
      * @param valueBytes Value bytes as read from the storage.
-     * @param valueHasTxId Whether the value has a transaction id prefix in it.
+     * @param valueHasTxData Whether the value has a transaction id prefix in it.
      * @return Binary row instance or {@code null} if value is a tombstone.
      */
-    private static @Nullable BinaryRow wrapValueIntoBinaryRow(byte[] valueBytes, boolean valueHasTxId) {
-        if (isTombstone(valueBytes, valueHasTxId)) {
+    private static @Nullable BinaryRow wrapValueIntoBinaryRow(byte[] valueBytes, boolean valueHasTxData) {
+        if (isTombstone(valueBytes, valueHasTxData)) {
             return null;
         }
 
-        return valueHasTxId
-                ? new ByteBufferRow(ByteBuffer.wrap(valueBytes).position(TX_ID_SIZE).slice().order(LITTLE_ENDIAN))
+        return valueHasTxData
+                ? new ByteBufferRow(ByteBuffer.wrap(valueBytes).position(VALUE_OFFSET).slice().order(LITTLE_ENDIAN))
                 : new ByteBufferRow(valueBytes);
     }
 
+    /**
+     * Converts raw byte array representation of the write-intent value into a read result adding newest commit timestamp if
+     * it is not {@code null}.
+     *
+     * @param valueBytes Value bytes as read from the storage.
+     * @param newestCommitTs Commit timestamp of the most recent committed write of this value.
+     * @return Read result instance or {@code null} if value is a tombstone.
+     */
+    private static ReadResult wrapUncommittedValue(byte[] valueBytes, @Nullable HybridTimestamp newestCommitTs) {
+        if (isTombstone(valueBytes, true)) {
+            return ReadResult.empty();
+        }
+
+        long txIdMsb = GridUnsafe.getLong(valueBytes, GridUnsafe.BYTE_ARR_OFF + TX_ID_OFFSET);
```

Comment:
You could extract `TxStateRocksDbStorage#bytesToUuid` into utils and introduce offset to it.
i believe there's also similar code in hash index storage for RocksDB